consensus/core/src/authority_service.rs

```diff
@@ -91,6 +91,22 @@ impl<C: CoreThreadDispatcher> NetworkService for AuthorityService<C> {
         }
         let peer_hostname = &self.context.committee.authority(peer).hostname;
 
+        // Reject blocks too advanced in round, to avoid overloading BlockManager and Synchronizer.
+        let highest_accepted_round = self.dag_state.read().highest_accepted_round();
+        // The threshold to
+        if highest_accepted_round + self.context.parameters.max_advance_round < signed_block.round()
+        {
```

Comment:
Probably that's extreme but _I think_ we might be susceptible to loss of liveness during an (extreme again) period of asynchrony (so we can't back fill earlier rounds as no commits happened for the past 100 rounds) and some weird combination of crashed nodes close to `f`? Ideally we would like to enforce the rule as long as the commits progress so we make sure that the [commit syncer](https://github.com/MystenLabs/sui/pull/17099) is running. Maybe we don't need to make things more complex now, but let's keep that under our radar

Reply:
Since BlockManager is not shared between threads, I chose another approach to drop incoming blocks once local commit is lagging quorum commit progress signficantly (500 commits by default).